src/Psalm/Internal/Analyzer/ClassLikeAnalyzer.php

```diff
@@ -596,10 +596,6 @@ public static function checkPropertyVisibility(
                 return $emit_issues ? null : true;
 
             case self::VISIBILITY_PROTECTED:
-                if ($appearing_property_class === $context->self) {
```

Comment:
This condition was flagged as redundant by this PR because it was previously checked here: https://github.com/vimeo/psalm/pull/5774/files#diff-218a87ce16087e317b1d9c674f9998fc9617b722fb0eeaa6d09c4d54032d7dbdL559

Reply:
Seems fine, it was added many eons ago and I can well believe it's redundant